src/ansys/fluent/core/fluent_connection.py

```diff
@@ -58,20 +59,33 @@ def run(self) -> None:
             cb()
 
 
-def get_container_ids() -> List[str]:
-    """Get a list of docker container IDs."""
+def get_container(container_id_or_name: str) -> Union[bool, Container, None]:
+    """Get the Docker container object.
+
+    Returns
+    -------
+    Union[bool, Container, None]
+        If the system is not correctly set up to run Docker containers, returns ``None``.
+        If the container was not found, returns ``False``.
+        If the container is found, returns the associated Docker container object.
+
+    Notes
+    -----
+    See `Docker container`_ for more information.
+
+    .. _Docker container: https://docker-py.readthedocs.io/en/stable/containers.html#docker.models.containers.Container
+
+    """
+    if not isinstance(container_id_or_name, str):
+        container_id_or_name = str(container_id_or_name)
     try:
-        logger.debug("Attempting to get docker container IDs...")
-        proc = subprocess.Popen(
-            ["docker", "ps", "-q"], stdout=subprocess.PIPE, stderr=subprocess.DEVNULL
-        )
-        output_bytes = proc.stdout.read()
-        output_str = output_bytes.decode()
-        ids = output_str.strip().split()
-    except:
-        logger.warning("Failed to get docker container IDs.")
-        ids = []
-    return ids
+        docker_client = docker.from_env()
+        container = docker_client.containers.get(container_id_or_name)
+    except docker.errors.NotFound:  # NotFound is a child from DockerException
+        return False
+    except docker.errors.DockerException:
+        return None
```

Comment:
probably better to propagate the exception than to differentiate between `None` and `False`

Reply:
@seanpearsonuk @mkundu1 My objective here, as in the function docstring, is: 

If the system is not correctly set up to run Docker containers, returns ``None``.
If the container was not found, returns ``False``.

Raising the exception would stop execution, while that is excessive when just checking if a container exists. This is used in a few places where we do not necessarily know whether Fluent is inside a container or not (e.g. when starting up FluentConnection without specifying when it is inside a container or not, or when checking from the watchdog whether Fluent is inside a container).

Reply:
On second thought, I think it is a good idea to at least log the exception, going to do that.